docs/installation/basic.md

```diff
@@ -15,28 +23,98 @@ Make sure you have the dependencies mentioned above installed before proceeding
+```sh
+git clone https://github.com/USERNAME/open-event-server.git
+cd open-event-server
+```
+
+* **Step 1** - Install python3 requirements. You need to be present in the root directory of the project.
+
+# System Wide Installation
+
+```sh
+sudo -H pip3 install -r requirements.txt
+```
+hint: You may need to upgrade your pip version and install following packages if you encounter errors while installing the requirements.
+
+**Note:** For Mac OS Sierra users, if you get an error that 'openssl/aes.h' could not be found when installing requirements.txt using pip, try the steps shown here - [OSX openssl header error](https://tutorials.technology/solved_errors/1-OSX-openssl_opensslv_h-file-not-found.html)
+
+# Installation in Virtual Environment
+
+You can use either **pip** or **pipenv** to install Open Event Server in a virtual environment.
+
+Firstly, open a terminal and enter
 
-* **Step 1** - Install Python 3 requirements.
+```sh
+# For linux users
+sudo apt-get install python3-dev
+sudo apt-get install libpq-dev
+sudo apt-get install libffi6 libffi-dev
+
+# For macOS users
+brew install python@3
+brew install libmagic
+```
+
+## Using pip and virtualenv
+
+Open a terminal and enter the following commands to setup a virtual environment
+
+```sh
+sudo apt-get install python3.7
+virtualenv -p python3.7 venv
+. venv/bin/activate
+```
+
+Now to install the dependencies using pip, type
 
 ```sh
 pip3 install -r requirements.txt
 ```
 
+## Using pipenv
+
+Using pipenv, you will not need to set up virtualenv. It will do it automatically for you
+
+To setup a virtual environment and install the dependices, enter in a terminal
+
+```sh
+pipenv --python 3.7.3 install
+```
+
+Now to activate the virtual environment, type
+
+```sh
+pipenv shell
+```
 
-* **Step 2** - Create the database. For that we first open the psql shell.
+* **Step 2** - Create the database. For that we first open the psql shell. Go to the directory where your postgres file is stored.
 
 ```sh
+# For linux users
 sudo -u postgres psql
+
+# For macOS users
+psql -d postgres
 ```
 
-* When inside psql, create a user for open-event and then using the user create the database.
+* When inside psql, create a user for open-event and then using the user create the database. Also, create a test database named opev_test for the test suites by dumping the oevent database into it. without this, the tests will not run locally.
+
+For ease of development, you should create Postgres user with the same username as your OS account. If your OS login account is _john_, for example, you should create _john_ user in Postgres. By this, you can skip entering password when using database.
 
 ```sql
 CREATE USER open_event_user WITH PASSWORD 'opev_pass';
 CREATE DATABASE oevent WITH OWNER open_event_user;
+CREATE DATABASE opev_test WITH OWNER open_event_user;
```

Comment:
Remove this step. Now we recommend docker container for testing

Reply:
Done